src/any_agent/tools/a2a.py

```diff
@@ -92,28 +95,46 @@ async def _send_query(
                 )
                 raise ValueError(msg)
 
-            if hasattr(response.root, "error"):
+            if isinstance(response.root, JSONRPCErrorResponse):
                 response_dict = {
                     "error": response.root.error.message,
                     "code": response.root.error.code,
                     "data": response.root.error.data,
                 }
-            elif hasattr(response.root, "result"):
-                response_dict = {
-                    "task_id": response.root.result.status.message.taskId,
-                    "context_id": response.root.result.status.message.contextId,
-                    "timestamp": response.root.result.status.timestamp,
-                    "status": response.root.result.status.state,
-                    "message": {
-                        " ".join(
-                            [
-                                part.root.text
-                                for part in response.root.result.status.message.parts
-                                if part.root.kind == "text"
-                            ]
-                        )
-                    },
-                }
+            elif isinstance(response.root, SendMessageSuccessResponse):
```

Comment:
The response can be a Message or a Task. @njbrake this built dict for Messages should make sense, but I guess you'll know more than me if a Message can reach us in practice instead of a Task.

Also, I'm replacing the checks about the kind with runtime `isinstance` type checks, which should be more reliable.

Reply:
Our a2a server would never return a message but another a2a server might.